plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java

```diff
@@ -46,13 +50,25 @@ protected void setup(Binder binder)
     @Singleton
     public static NessieIcebergClient createNessieIcebergClient(IcebergNessieCatalogConfig icebergNessieCatalogConfig)
     {
+        boolean isBearerAuth = icebergNessieCatalogConfig.getSecurity().filter(isEqual(BEARER)).isPresent();
+        checkState(
+                isBearerAuth || icebergNessieCatalogConfig.getBearerToken().isEmpty(),
+                "'iceberg.nessie-catalog.authentication.token' must be configured only with 'iceberg.nessie-catalog.authentication.type' BEARER");
+        checkState(
+                !isBearerAuth || icebergNessieCatalogConfig.getBearerToken().isPresent(),
+                "'iceberg.nessie-catalog.authentication.token' must be configured with 'iceberg.nessie-catalog.authentication.type' BEARER");
```

Comment:
We can verify invalid properties within IcebergNessieCatalogConfig. You can refer to MySqlJdbcConfig.

Reply:
done